### ISSUE TITLE
[$500]  Task - Hand cursor is displayed when hovering over the task description placeholder

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#35207]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.32-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open a chat
2. Go to the assign task
3. Hover over the description

## Expected Result:
Hand cursor should not be displayed when hovering over the task description placeholder

## Actual Result:
Hand cursor is displayed when hovering over the task description placeholder

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/ccfb2836-1316-4597-971f-81b428561206

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bb349ec7e874f74f</li>
        <li>Upwork Job ID: 1750652923015548928</li>
        <li>Last Price Increase: 2024-02-01</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by ikevin127** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue

Hand cursor is displayed when hovering over the task description placeholder.

### What is the root cause of that problem?

We're using cursor: pointer on the task description label.

### What changes do you think we should make in order to solve the problem?

We should change the style to cursor: default.

**Comment by dragnoir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Wrong cursor displayed hovering description top area.

### What is the root cause of that problem?

To understand what's happening, let's add a green background to the View wrapping the `<RNTextInput>` component.

![image](https://github.com/Expensify/App/assets/12425932/f0276954-17b3-45e6-9b93-355f13c26c2a)

As on the screenshot, there's another layer at the top. This is because `styles.textInputMultilineContainer` is adding an extra padding. 

![image](https://github.com/Expensify/App/assets/12425932/ffa54902-202c-482f-906a-c48a30d6ac7d)

When this padding is added, the extra zone (padding/light green on screenshot above) will ask about the style of the cursor state on hover. 
De we deliver it? NO, then this extra zone will check parents for the cursor state and because the 1st pârent with the state set is `PressableWithoutFeedback` then the cursos will be `pointer`

### What changes do you think we should make in order to solve the problem?

Where we set `styles.textInputMultilineContainer` , we need to make sure of the cursos state on hover with `styles.cursorText`

```diff
<View style={[
  styles.textInputAndIconContainer, 
  isMultiline && hasLabel && styles.textInputMultilineContainer, 
  styles.pointerEventsBoxNone
+ styles.cursorText
]}>
```

### POC:

https://github.com/Expensify/App/assets/12425932/d16fdf09-f81e-45b9-981c-ac83e635be57

**Comment by abekkala** [comment]
@rushatgabhane do you have any feedback for the proposals above?

**Comment by rushatgabhane** [comment]
reviewing

**Comment by ikevin127** [comment]
Just wanted to mention my proposal vs the second one's timestamp snapshots to avoid any confusions as my colleague mentioned in the beginning.

In chronological order:

**ikevin127**: Jan 26, 2024, 12:57 AM GMT+2 - initial proposal timestamp
**Krishna2323**: Jan 26, 2024, 1:01 AM GMT+2 - initial proposal timestamp
**Krishna2323**: between 1:01 and 1:22 - had 4 proposal edits, his last one at 1:22 looking like this:
<details>
<summary>Krishna2323's 4th edit @ 1:22</summary>

<img width="657" alt="Screenshot 2024-01-31 at 14 17 42" src="https://github.com/Expensify/App/assets/56457735/9fbf404f-d359-407a-8d79-67a6fd629119">

</details>

**ikevin127**: Jan 26, 2024, 1:28 AM GMT+2 - 1st edit of proposal timestamp - added more context to the initial proposal, being **first to mention the full RCA** (why do we see the pointer cursor and where is the style coming from)
**ikevin127**: Jan 26, 2024, 1:33 AM GMT+2 - 2nd edit of proposal timestamp - I added video of the solution result

**Comment by ikevin127** [comment]
Correction: your initial proposal had the incomplete RCA 🙂

**Comment by ikevin127** [comment]
Incorrect, if you take a closer look at the issue's title, actual result and expected result - one can clearly deduce that when the proposal's template `What is the root cause of that problem?` question is asked, the expected RCA here would be to have a clear understanding of why we see the hand (pointer) cursor.

Regarding your argument, I think that only mentioning the padding does not fulfil this analysis IMO as that's just the reason that causes the hand cursor to be seen, without any mention of where is that styling coming from. This way you'll apply a fix to something you don't fully understand.

We all know that without a clear understanding of the root cause, a solution cannot be applied - no matter whether or not it looks good or it fixes the issue (apparently). See example: https://github.com/Expensify/App/issues/35197#issuecomment-1914614538

**Comment by rushatgabhane** [comment]
The solution is simple but I'm not sure which proposal to select because of edits

**Comment by rushatgabhane** [comment]
> We should change the style to cursor: default.


@ikevin127 i don't think we should change default cursor type of BaseGenericPressable. Because a Pressable is meant to be clicked and should have the hand cursor

**Comment by rushatgabhane** [comment]
I think we should go with @Krishna2323's proposal https://github.com/Expensify/App/issues/35207#issuecomment-1911133220 it seems straightforward to me.


🎀 👀 🎀

**Comment by ikevin127** [comment]
@rushatgabhane I see wehat you're saying, but in this case we're talking about an input that is being wrapped by a pressable.

As mentioned in my solution, because we're not passing the input's role to the wrapping pressable - this inherits the pressable default cursor which is pointer, but in our case we don't need that since it wraps an input.

The input itself has the cursor text already - the only issue here is that because of the padding, when we hover above the input we see the pressable default cursor which is pointer because we didn't pass the input's role to the pressable.

If the role is passed, the pressable already has the logic to set cursor default - only issue is we currently don't pass the role 😁

**Comment by rushatgabhane** [comment]
@ikevin127 could you please help me understand what `role` is ?

**Comment by ikevin127** [comment]
@rushatgabhane Sure, here's a code block for better visualisation:

https://github.com/Expensify/App/blob/43d1576cab735b64eb1c8153b0cd06a1a5128041/src/components/Pressable/GenericPressable/BaseGenericPressable.tsx#L64-L75

As the comment says, there can be different roles for a Pressable and based on the role that function determines what cursor the pressable will display.

Our input has the `presentation` role but because it's not passed to its wrapping Pressable - the cursor defaults to pointer as a Pressable should.

**Comment by ikevin127** [comment]
I think it's a more holistic solution because it solves the problem from the root instead of overriding the inherited Pressable cursor style which is incorrect due to the fact that we didn't pass the input role to the Pressable ✌️

**Comment by ikevin127** [comment]
Incorrect!

Expected result states:

> Expected Result:

> Hand cursor should not be displayed when hovering over the task description placeholder

The root cause which differs from your proposal and I mentioned that in mine is complete since it identifies what causes the hand cursor, it's not just the padding.

@rushatgabhane  This is why I'm stating that my solution is also completely different than the selected one since it fixes the problem from the root cause - doesn't just put a bandaid on it by overriding the Pressable styles which shouldn't even be there in the first place but they are because the input role wasn't passed to its wrapping Pressable.

**Comment by abekkala** [comment]
@aldo-expensify can you review this please?

**Comment by aldo-expensify** [comment]
> @ikevin127 i don't think we should change default cursor type of BaseGenericPressable. Because a Pressable is meant to be clicked and should have the hand cursor

I see your point here, but in this case it doesn't make sense to me to try to preserve the `pointer` cursor of the "pressable" if we really don't want to show it when it wraps an input. Also, I would assume `BaseGenericPressable` has the `role` prop just for cases like this.

Playing with the padding so it becomes impossible to hover just over the "pressable" feels more brittle to me than just aligning the role of the "pressable" with the input that is wrapping.

I prefer @ikevin127 's [proposal ](https://github.com/Expensify/App/issues/35207#issuecomment-1911128428) that adds `role={inputProps.role}` because it is simple and more robust in my opinion

@Krishna2323 's proposal is also simple, but I like it better the use of `role` to solve the issue.

@ikevin127 won by timestamp on posting a proposal first, but that first proposal is veeery generic:
- The RCA is just guess anyone could take based on the pointer cursor appearing, and
- proposed fix is very generic just like the RCA

<img width="708" alt="image" src="https://github.com/Expensify/App/assets/87341702/455ecca9-a93a-4560-9579-6afd89cf1c6d">


In summary, the way I see things is:

- @ikevin127 posted first a very generic proposal with little specific details that I would have rejected
- @Krishna2323 posted a proposal with a RCA that shows to me that an investigation did happen. I would consider this the first real proposal
- @ikevin127 and @Krishna2323 continued to polish their proposals (I assume that @ikevin127 used @Krishna2323 's findings)
- I consider the current proposals RCAs the same and consider that @Krishna2323 found it first
- I'm see @ikevin127 's solution different enough from @Krishna2323 's solution to consider it as a different proposal
- I like more @ikevin127 's `role` solution

**Comment by aldo-expensify** [comment]
About other proposals:
- @tienifr 's proposed solution based on adjusting padding, which is something I would prefer to avoid
- @dragnoir 's is a bit different, but I still prefer fixing this using `role`. I do appreciate though the images to show the padding issue, they are helpful to visualize the problem!

**Comment by aldo-expensify** [comment]
Oops, I thought your proposal was to adjust paddings hehe, I see now that you are proposing change the cursor style. Re-reading again..................

**Comment by ikevin127** [comment]
I think @aldo-expensify's assessment is fair here. I was amazed that @rushatgabhane did not go with the `role` based solution once I explained that we have that prop on Pressables for a reason.

My RCA identified the full root cause (why we see the hand cursor, where is the style coming from) and came up with the correct solution which is to use the Pressable's role which will return text cursor if passd, instead of adding that style directly where the other proposal suggested - this would mean that the Pressable will still have the pointer cursor, we would just override it from a child of the Pressable.

**Comment by ikevin127** [comment]
> does applying role wouldn't override the cursor style of the parent component? Does role wouldn't override pressable's default cursor style?

Nope, this is the point I'm trying to make here - since the Pressable is the parent that wraps the input but we're not passing the `role` of the input to the Pressable, its function here will return the pointer style cursor:

https://github.com/Expensify/App/blob/43d1576cab735b64eb1c8153b0cd06a1a5128041/src/components/Pressable/GenericPressable/BaseGenericPressable.tsx#L64-L75

Passing the input `role` to the Pressable would simply **return** the correct style which is cursor text in this case.

I don't understand how can somebody call this overriding ?
Overriding would be to set the styles as the other proposal suggested, directly to the children of the Pressable, while the Pressable still returns cursor pointer because the `role` wasn't passed.

No matter how much you throw accusations, call my clarifications redundant and that I'm trying to create confusion - this is still the reality of what we're dealing with here in this issue and we should'n just slap a bandaid on something when we don't fully understand the workings of the code.

Try to keep the conversation without any personal comments, take this as a piece of feedback.
Such comments are not necessary ✌

**Comment by aldo-expensify** [comment]
Drawing the line wether a proposal is different than another or not is pretty subjective, so I think we have to accept that what was decided may not be aligned with our opinion. It is true that all proposals in the end are changing a cursor style somehow, but that is unavoidable, so I don't think `I still don't think adding role is different from adding styles on view` is a strong argument here, and I don't think sharing the same RCA is reason enough either.

For me, the implementation using `role` is cleaner, it is a prop that we already have and seems to be exactly for what we want here. Adding `cursor:'text'/styles.cursorText ` to `textInputMultilineContainer` doesn't align with its sibling style `textInputContainer`. I also believe there is a difference between correcting the style at the source compare to correcting it in a child component (correcting it at the source is better for me).

> your proposal was not selected because you copied, sorry for being rude but you are trying to put redundant arguments to create confusions

Again, there is lots of subjective things in this conversation and decision, and I don't think this comment above is accurate neither constructive, so I would suggest avoiding such comments because it derails conversations into arguments that become personal (this does look like a personal attack to me). Keeping a good and friendly work environment is important for a long term working relationship, so I would recommend to not lose sight of that.

I think everything that was important to be said about why one proposal is better than the other has been said, the latest comments in this thread have very little value, so I ask you to refrain from continuing this discussion.

**Comment by aldo-expensify** [comment]
@ikevin127 refrain with reacting with 👎 or such because it is not constructive and may be annoying to the commenter (or me reading it)

@Krishna2323 are you suggesting that we should split 250/250 because you contributed here and the other issues is an example of this outcome?

If that is your suggestion, I will decline that because: it is true that sometimes we do that, but I don't think that this should be the norm. I don't want everyone requesting this just because they posted the right RCA before or because in their opinion their solution is the same.

This is just how the rules currently are, sometimes you will spend time investigating and you just won't get rewarded, but hopefully some other issue reward will make up for it so you still feel it was worth it.

If you think the system should be improved, I think you should bring the conversation to the right slack channel.

>  despite the fact that the other two proposals were incomplete and likely to cause regressions

I disagree with that they would likely cause regressions

> Regardless, I'll consider it unfortunate luck for me.

I think this is the best in this case, there is a lot in this that is out of your control. For example, if there was another internal engineer assigned, that engineer could have chosen your solution (same as @rushatgabhane) as best. Just keep in mind that I don't have a personal preference between you or @ikevin127 , I don't know any of you personally, I'm just making a decision considering what I think is fair and what I think is best for our codebase. Hopefully, in average, you shouldn't feel wronged because sometimes you will get lucky and sometimes someone else.

**Comment by ikevin127** [comment]
PR https://github.com/Expensify/App/pull/36286 ready for review! 🚀

> [!NOTE]
> I passed down the `role` prop for the `BaseTextInput/index.native.tsx` too, for good measure ✌

**Comment by ikevin127** [comment]
Looks like the issue was fixed by this recently merged PR https://github.com/Expensify/App/pull/35878 which was just deployed on staging, coming from this issue: https://github.com/Expensify/App/issues/33893. I closed the PR and I think this issue is resolved 👏

cc @abekkala

**Comment by aldo-expensify** [comment]
> I closed the PR and I think this issue is resolved 👏

Yeah, I checked too, it seems to be working fine now

**Comment by aldo-expensify** [comment]
Thank you all for work!
